### PR TITLE
Add automatic secret generation to New-AzureRmSPCredential

### PR DESCRIPTION
**Pull request by maddieclayton**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/NewAzureADServicePrincipalCommand.cs [hunk 1]
```diff
@@ -362,7 +366,8 @@ private void CreateSimpleServicePrincipal()
                                                       "having AppId '{0}' with '{1}' role over scope '{2}'.", createParameters.ApplicationId, this.Role, this.Scope);
             if (ShouldProcess(target: createParameters.ApplicationId.ToString(), action: shouldProcessMessage))
             {
-                var servicePrincipal = ActiveDirectoryClient.CreateServicePrincipal(createParameters);
+                PSADServicePrincipalWrapper servicePrincipal = new PSADServicePrincipalWrapper(ActiveDirectoryClient.CreateServicePrincipal(createParameters));
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/NewAzureADServicePrincipalCommand.cs]: does cmdlet outputtype attribute need to change?

File: src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/NewAzureADSpCredentialCommand.cs [hunk 2]
```diff
@@ -132,7 +137,24 @@ public override void ExecuteCmdlet()
                 }
                 else
                 {
-                    throw new InvalidOperationException("No valid keyCredential or passwordCredential to update!!");
+                    // If no credentials provided, set the password to a randomly generated GUID
+                    Password = Guid.NewGuid().ToString().ConvertToSecureString();
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/NewAzureADSpCredentialCommand.cs]: How easy would it be to add tests for the new parameter set for new-spcredential, and to update the tests for both cmdlets to ensure the wrapper is filled out?

Comment (markcowl) [reply src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/NewAzureADSpCredentialCommand.cs]: If we can't add automation this milestone, please file an issue for adding automation next milestone

**Comment by maddieclayton** [comment]
I will try to get tests in a different PR before we release, but opened an issue here (https://github.com/Azure/azure-powershell/issues/7798) so we can merge this PR.

File: src/ResourceManager/Resources/Commands.Resources/Models.Authorization/PSADCredentialWrapper.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿using Microsoft.Azure.Graph.RBAC.Version1_6.ActiveDirectory;
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources/Models.Authorization/PSADCredentialWrapper.cs]: @maddieclayton please add the license header to this file

File: src/ResourceManager/Resources/Commands.Resources/Models.Authorization/PSADServicePrincipalWrapper.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.Azure.Graph.RBAC.Version1_6.ActiveDirectory;
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources/Models.Authorization/PSADServicePrincipalWrapper.cs]: @maddieclayton same comment here about license header

File: src/ResourceManager/Resources/Commands.Resources/Models.Authorization/PSADCredentialWrapper.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+using System.Security;
+
+namespace Microsoft.Azure.Commands.Resources.Models.Authorization
+{
+    public class PSADCredentialWrapper : PSADCredential
+    {
+        public PSADCredentialWrapper(PSADCredential cred)
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources/Models.Authorization/PSADCredentialWrapper.cs]: @maddieclayton I would add a null-check at the beginning of this constructor to be safe

File: src/ResourceManager/Resources/Commands.Resources/Models.Authorization/PSADServicePrincipalWrapper.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+using System.Security;
+
+namespace Microsoft.Azure.Commands.Resources.Models.Authorization
+{
+    public class PSADServicePrincipalWrapper : PSADServicePrincipal
+    {
+        public PSADServicePrincipalWrapper(PSADServicePrincipal sp)
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources/Models.Authorization/PSADServicePrincipalWrapper.cs]: @maddieclayton same comment about adding a null-check

File: src/ResourceManager/Resources/Commands.Resources/help/New-AzureRmADServicePrincipal.md [hunk 7]
```diff
@@ -585,6 +585,8 @@ Parameters: ApplicationObject (ByValue)
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources/help/New-AzureRmADServicePrincipal.md]: @maddieclayton we may want to add an example to both cmdlets for the new scenario and show the output containing the password

**Pull request merged**